src/services/dashboards/widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrend.vue

```diff
@@ -0,0 +1,261 @@
+import type { WidgetProps, WidgetExpose } from '@/services/dashboards/widgets/config';
+import { CHART_TYPE, WIDGET_SIZE } from '@/services/dashboards/widgets/config';
+import type { HistoryDataModel, Legend, XYChartData } from '@/services/dashboards/widgets/type';
+import { useWidgetFrameProps } from '@/services/dashboards/widgets/use-widget-frame-props';
+import { useWidgetLifecycle } from '@/services/dashboards/widgets/use-widget-lifecycle';
+// eslint-disable-next-line import/no-cycle
+import { useWidgetState } from '@/services/dashboards/widgets/use-widget-state';
+import {
+    getDateAxisSettings,
+    getLegends,
+    getRefinedXYChartData,
+} from '@/services/dashboards/widgets/widget-chart-helper';
+import {
+    getWidgetTableDateFields, sortTableDataByDate,
+} from '@/services/dashboards/widgets/widget-table-helper';
+
+const DATE_FORMAT = 'YYYY-MM';
+const DATE_FIELD_NAME = 'date';
+const USAGE_SOURCE_UNIT = 'GB';
+const SELECTOR_TYPE_FIELDS_KEY_MAP = {
+    cost: 'usd_cost',
+    usage: 'usage_quantity',
+};
+const USAGE_TYPE_LABEL_MAP = {
+    'data-transfer.out': 'Transfer-out',
+    'data-transfer.in': 'Transfer-in',
+    'data-transfer.etc': 'etc.',
+};
+
+const props = defineProps<WidgetProps>();
+
+const chartContext = ref<HTMLElement|null>(null);
+const chartHelper = useAmcharts5(chartContext);
+
+const state = reactive({
+    ...toRefs(useWidgetState<HistoryDataModel['results']>(props)),
+    fieldsKey: computed<string>(() => SELECTOR_TYPE_FIELDS_KEY_MAP[state.selectedSelectorType]),
+    chart: null as null | XYChart,
+    chartData: computed<XYChartData[]>(() => {
+        const valueKey = `${state.fieldsKey}_sum`;
+        return getRefinedXYChartData(state.data, state.groupBy, DATE_FIELD_NAME, valueKey);
+    }),
+    tableFields: computed<Field[]>(() => {
+        if (!state.groupBy) return [];
+        const _textOptions: Field['textOptions'] = {
+            type: state.fieldsKey === 'usd_cost' ? 'cost' : 'size',
+            sourceUnit: 'GB',
+        };
+        const refinedFields = getWidgetTableDateFields(state.granularity, state.dateRange, _textOptions, state.fieldsKey);
+        return [
+            { label: ' ', name: state.groupBy },
+            ...refinedFields,
+        ];
+    }),
+    dateRange: computed<DateRange>(() => {
+        const end = state.settings?.date_range?.end ?? dayjs.utc().format(DATE_FORMAT);
+        const range = props.size === WIDGET_SIZE.full ? 11 : 3;
+        const start = dayjs.utc(end).subtract(range, 'month').format(DATE_FORMAT);
+        return { start, end };
+    }),
+    legends: [] as Legend[],
+});
+const widgetFrameProps:ComputedRef = useWidgetFrameProps(props, state);
+
+/* Util */
+const getRefinedTableData = (results: HistoryDataModel['results']): HistoryDataModel['results'] => results.map((result) => ({
+    ...result,
+    [state.groupBy]: USAGE_TYPE_LABEL_MAP[result[state.groupBy]],
+}));
```

Comment:
description: This changes `transfer-data.out` to `Transfer-out`. This converting is necessary because there's no reference data or else🥲